apps/extract-stack/src/extract-namespace-members.ts

```diff
@@ -0,0 +1,145 @@
+  const [userOauthAccessTokenPayload, ...rest] = await clerkClient.users.getUserOauthAccessToken(userId, forgeryIdProvider);
+  if (!userOauthAccessTokenPayload) throw new Error("Failed to get token");
+  if (rest.length !== 0) throw new Error("wtf ?");
+
+  return userOauthAccessTokenPayload.token;
+}
+
+const initSourceControl = async (userId: string, sourceControl: 'github' | 'gitlab') => {
+  const accessToken = await fetchSourceControlAccessToken(userId, `oauth_${sourceControl}`);
+  if (sourceControl === 'github') return new GitHubSourceControl(accessToken);
+  if (sourceControl === 'gitlab') return new GitlabSourceControl(accessToken);
+  return null;
+}
+
+const db = drizzle(client);
+
+const context: Context<GetNamespaceMembersSourceControl, GetNamespaceMembersEntities> = {
+  entities: {
+    members,
+    repositoriesToMembers
+  },
+  integrations: {
+    sourceControl: null,
+  },
+  db,
+};
+
+type ExtractNamespaceMembersPageInput = {
+  namespace: Namespace;
+  repositoryId: Repository['id'];
+  sourceControl: "github" | "gitlab";
+  userId: string;
+  paginationInfo?: Pagination;
+}
+
+const extractNamespaceMembersPage = async ({ namespace, repositoryId, sourceControl, userId, paginationInfo }: ExtractNamespaceMembersPageInput) => {
+  const page = paginationInfo?.page;
+  const perPage = paginationInfo?.perPage;
+
+  context.integrations.sourceControl = await initSourceControl(userId, sourceControl);
+
+  const { members, paginationInfo: resultPaginationInfo } = await getNamespaceMembers({
+    externalNamespaceId: namespace.externalId,
+    namespaceName: namespace.name,
+    repositoryId,
+    perPage: perPage,
+    page: page
+  }, context);
+
+  await extractMembersEvent.publish({
+    memberIds: members.map(member => member.id)
+  }, {
+    version: 1,
+    caller: 'extract-namespace-member',
+    sourceControl: sourceControl,
+    userId: userId,
+    timestamp: new Date().getTime(),
+  });
+
+
+  return { members, pagination: resultPaginationInfo };
+};
+
+export const eventHandler = EventHandler(extractRepositoryEvent, async (ev) => {
+  const repository = await db.select().from(repositories).where(eq(repositories.id, ev.properties.repositoryId)).get();
+  const namespace = await db.select().from(namespaces).where(eq(namespaces.id, ev.properties.namespaceId)).get();
+
+  if (!repository) throw new Error("invalid repo id");
+  if (!namespace) throw new Error("Invalid namespace id");
+
+  const { pagination } = await extractNamespaceMembersPage({
+    namespace: namespace,
+    repositoryId: ev.properties.repositoryId,
+    sourceControl: ev.metadata.sourceControl,
+    userId: ev.metadata.userId,
+  });
+
+  const arrayOfExtractMemberPageMessageContent = []; 
+  for (let i = 2; i <= pagination.totalPages; i++) {
+    arrayOfExtractMemberPageMessageContent.push({
+      namespace,
+      repositoryId: ev.properties.repositoryId,
```

Comment:
why do you send repository by id but namespace by entity ?

EDIT: nevermind i get it